modules/packages/LocalAtomics.chpl

```diff
@@ -0,0 +1,382 @@
+/*
+ * Obtained from https://github.com/LouisJenkinsCS/LocalAtomics/blob/master/src/LocalAtomics.chpl
+ */
+
+module LocalAtomics {
+  /*
+     Planned usage:
+     var head : LocalAtomicObject(unmanaged Node(int));
+  // Adding a new node...
+  head.write(new unmanaged Node(int)); // Need 'write(objType)'
+  */
+
+  extern {
```

Comment:
I think that `CMPXCHG16B` stuff should be added directly to the runtime and `extern` from that, what do you think @mppf and @gbtitus? Right now it would require `extern` blocks and `CHPL_LLVM=llvm`, but its not really necessary if these functions are already available from the runtime.

Reply:
The runtime could provide it, certainly, perhaps in files that were siblings of `include/atomics/*/chpl-atomics.h`.  I don't think that's strictly necessary for the purposes of this PR, however.  That said, if you were going to support ARM you'd obviously need to update what's here so as to use a different `_asm` block for that architecture.

Reply:
Okay, I just disliked the fact that you need to compile with `llvm` to use this module, when its not strictly necessary; LLVM takes a long time to build, and I know that there are some platforms that have difficulty building it (as I've run into issue when trying to build it myself). I'd like to commit to a quick runtime change, if possible.

Reply:
I think it'd make sense to add some minimal 128-bit CAS support to the runtime.

Reply:
Is `CHPL_LLVM=llvm` set by default now? I've noticed that I do not have `CHPL_LLVM` environment variable set; if so, then it doesn't matter if the `LocalAtomics.chpl` module has an `extern` block.